### ISSUE TITLE
Copy image to clipboard [Feature]

### ISSUE THREAD

**Issue by lieszkol** [typora/typora-issues#4476]
I can copy an image from a browser and paste it into Typora. However, if I want to copy an image from one document in Typora to another, this does not work, whether I try the "copy" or the "copy image" command from the context menu. I also cannot copy an image from Typora and, say, paste it into photoshop or some other program. This is on Windows 7 with a png image, but other formats don't work either.

**Comment by Pandaa-P** [comment]
On Windows 10, I tested both copy and copy image, image formats [JPG, PNG. GIF and WEBP]. 

- I can copy from typora to another document in typora ✔
    - typora <----> typora: from typora to typora work for me ✔
- I can't copy from typora to other programs ❌

**Comment by lieszkol** [comment]
Please try to copy an image from one document to another document that is in a different folder. 

- If you use "Copy" it pastes the relative path of the image and it doesn't show up. Here is an example: 

`![image-20210616161654541](.Tipton 2021/image-20210616161654541.png)`

- If you use "Copy Image" it then pastes an absolute path - which works, as long as the actual image file remains in the same location. Here is an example:

`![image-20210615230842475](file://G:\Szelence\KB\.Mana%202021\image-20210615230842475.png?lastModify=1624010363)`

Typora ought to copy the image contents, not a path to the image file. "Copy-Paste" implies creating a copy of an object. When you paste it into a different document in Typora, it should treat it the same way as when you copy an image from your browser. It should save a copy of the image to the filesystem (in accordance with the image handling preferences set in Typora).

Here are my Typora settings for image handling:
- When Insert... Copy Image to Custom Folder
- Custom folder: ./.${filename}
- Apply above rules to local images CHECKED
- Apply above rules to online images CHECKED
- Use relative path if possible CHECKED

Note: I use the "custom folder" option so that image files don't show up in the Typora sidebar.

**Comment by Pandaa-P** [comment]
I re-tested in typora and it seems it can't copy the contents of the image but it only copy the path. 
#### I replicated your settings on windows 10 
- Copy Image to Custom Folder ✔
- Custom folder: ./.${filename} `F:\.Tipton 2021` ✔
- Apply above rules to local images ✔
- Apply above rules to online images ✔
- Use relative path if possible ✔

#### What I can and can not do:
Can 
- Copy image from one document to another in the same folder ✔
- Copy image from one document to another in a different folder ✔
- Copy image from one document to another in a different folder in another drive ✔
- Copy the image to another program ( Photoshop, ms paint) ❌ 
 - Typora can only copy the path of the image from one document to another that why it fail even though you can 
    - Copy image from any drive in PC to Typora, and it would show up in Typora
    - Copy the image (not the URL) from any browser to Typora, and it it would show up in Typora

#### What in windows 10 clipboard contents ?

#### With `copy` command 

- in Typora `![test-img](F:/.Tipton 2021/test-img.png)`
- in other text based programs (notepad, word) `![test-img](F:/.Tipton 2021/test-img.png)`
- in other image based programs (Photoshop, ms paint) `nothing`

#### With `copy image` command
- in Typora `![test-img](file://F:\.Tipton%202021\test-img.png?lastModify=1624197802)`
- in other text based programs (notepad, word) `![test-img](file://F:\.Tipton%202021\test-img.png?lastModify=1624197802)`
- in other image based programs (Photoshop, ms paint) `nothing`


TL;DR
- Typora can only copy the path from inside Typora instances, it can't copy the image itself.
- It can copy the image from the disk and the browser but can't do the same from Typora to other programs

**Comment by Pandaa-P** [comment]
your settings should solve the problem since every image would be copied with absolute path I don't know why with you have problem with it with these setting, on the other hand here typora with the default setting using the relative path of image.

![Capture2](https://user-images.githubusercontent.com/85793213/122677923-1f72be80-d1ed-11eb-9a0f-d267756495e5.PNG)

**Comment by lieszkol** [comment]
Thank you for confirming my description of this feature request, we are in agreement. As I wrote in my previous comment, copying the absolute path _"works, as long as the actual image file remains in the same location."_ There are many reasons why the original image may no longer be in the same location. Say, you have a Typora folder on a pen drive, and a Typora folder on your PC. You can't just copy articles with images from the pen drive to your PC, since once you remove the pen drive, the images no longer show up in the articles on your PC. Basically because of this, you can no longer move articles around in the file system (by moving the folders around), since the images aren't in the same folder as the articles. This is very frustrating, it prevents the user from re-organizing their notes. 

As I indicated in the title, this is a feature request. I don't know if there is some fundamental limit with electron apps that makes it impossible to do what I am asking for (copy the image contents to the clipboard instead of the image path). I assume not, since Typora can copy text to the clipboard, and Typora surely can read the contents of an image file from disk. 

In any case, I believe this would be a useful "feature" to include in Typora, after all, this is effectively how classic programs like Microsoft Word have worked for over two decades (if you copy an image from one word document to another, word doesn't paste a link to the original image, it pastes the image itself). Perhaps not the best analogy (since word embeds images, whereas a Markdown editor like Typora of course stores images in the file system), but from a user's perspective, the mechanism is similar. Copy-paste means create a copy of the object and place that copy into the target document/filesystem/email etc.

Thank you for considering my feature request, and thank you for Typora!

**Comment by abnerlee** [comment]
If text is selected, text will be put into clipboard, even when it describes an image in markdown.

There is a "copy image" command in context menu which will copy the image content, that command has been fixed in latest version.

**Comment by alexanderwelbourne** [comment]
This is an issue for me again on Version 1.10.9 (7393) on Mac. If I "copy as image" from the context window I can only paste within typora windows, it doesn't work externally.
